### ISSUE TITLE
[BUG] Unable to add pages with future starttime to index queue

### ISSUE THREAD

**Issue by DanielSiepmann** [TYPO3-Solr/ext-solr#3547]
**Describe the bug**
Pages with an `starttime` set to the future are not indexed, until they are modified after reaching starttime, or reindexed via scheduler after reaching starttime.

**To Reproduce**
1. Create a new page
2. Set `starttime` to the future
3. Save page
4. Inspect the table `tx_solr_indexqueue_item`, you won't find an entry for the page,.

**Expected behavior**
I would expect to create or edit an TYPO3 page record with a `starttime` in the future. This page should then be added as index queue item to the database table to be indexed as soon as the page is available.

**Screenshots**
No Screenshots available

**Used versions (please complete the following information):**
 - TYPO3 Version: e.g. 11.5.24
 - Browser: Firefox, but should not be related.
 - EXT:solr Version: 11.5.1
 - Used Apache Solr Version: 8.11.1, but should not be related
 - PHP Version: 8.1.16, but should not be related
 - MySQL Version: 8.0.32-0ubuntu0.20.04.2. but should not be related

**Additional context**
We already wrote about that on slack: https://typo3.slack.com/archives/C02FF05Q4/p1678968753747799

Related Code:
A TSFE will be initialized prior adding the entry to the DB
That initialization will fail. The TSFE can not find the page as it is not available yet at https://github.com/TYPO3/typo3/blob/v11.5.24/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php#L1084. This will result in an Exception at https://github.com/TYPO3/typo3/blob/v11.5.24/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php#L998. This then leads to missing TSFE: https://github.com/TYPO3-Solr/ext-solr/blob/11.5.1/Classes/FrontendEnvironment/Tsfe.php#L186 which then will prevent further processing and adding the db entry.

**Comment by DanielSiepmann** [comment]
Our current hotfix for our project:

```patch
diff --git a/Classes/FrontendEnvironment.php b/Classes/FrontendEnvironment.php
index 80ae2c67..88077e0a 100644
--- a/Classes/FrontendEnvironment.php
+++ b/Classes/FrontendEnvironment.php
@@ -74,11 +74,11 @@ class FrontendEnvironment implements SingletonInterface
             $rootPageRecordUid = $pageRecord['l10n_parent'];
         }
 
-        $tsfe = GeneralUtility::makeInstance(Tsfe::class)->getTsfeByPageIdIgnoringLanguage($rootPageRecordUid);
+        $tsfe = GeneralUtility::makeInstance(Tsfe::class)->getTsfeByPageIdIgnoringLanguage(1);
         if (!$tsfe instanceof TypoScriptFrontendController) {
             return false;
         }
-        $configuration = $this->getConfigurationFromPageId($rootPageRecordUid, '', $tsfe->getLanguage()->getLanguageId());
+        $configuration = $this->getConfigurationFromPageId(1, '', $tsfe->getLanguage()->getLanguageId());
         $allowedPageTypes = $configuration->getIndexQueueAllowedPageTypesArrayByConfigurationName($configurationName);
         return in_array($pageRecord['doktype'], $allowedPageTypes);
     }
```

Our root page has uid 1. We always fetch TSFE for that page. That works as we don't have additional TypoScript on sub pages. I don't see an easy proper fix but still wanted to share our hotfix for a single project.

This fix allows the page to be added to the queue. As no exception occurs while building TSFE in order to fetch TypoScript. Everything else works as expected already. The item from queue will only be processed once it is available.

**Comment by dmitryd** [comment]
Note that this PR will also fix #3702

**Comment by dogawaf** [comment]
Hi

I can still reproduce the issue: 

- Create a new page
- Set starttime to the future
- Save page
- Inspect the table tx_solr_indexqueue_item, you won't find an entry for the page,.

Reproduced on: 

- TYPO3 11.5, Solr 11.5.5, delayed or immediate
- TYPO3 12.4, Solr 12.0.5, delayed or immediate

Maybe a regression was introduce since the fix?

With a small modification to the method `ConfigurationAwareRecordService->getRecordForIndexConfigurationIsValid`:

```php
$queryBuilder = $this->getQueryBuilderForTable($recordTable);
$queryBuilder->getRestrictions()->removeByType(StartTimeRestriction::class); // <---- here we remove the starttime restriction

```
The event is now added to the indexing, with the `changed` column corresponding to the starttime in the future, so it will only be indexed after that date.
